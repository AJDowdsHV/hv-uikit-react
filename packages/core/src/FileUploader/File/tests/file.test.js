/*
 * Copyright 2019 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* eslint-env jest */

import React from "react";
import { mount } from "enzyme";

import FileWithStyles from "..";
import HvProvider from "../../../Provider";

let wrapper;

const dataFail = {
  name: "somefile.png",
  size: 875,
  errorMessage: "File type not allowed for upload",
  progress: 25,
  status: "fail",
  fileStatus: "fail"
};

const dataSuccess = {
  name: "somefile.png",
  size: 875,
  errorMessage: "File type not allowed for upload",
  progress: 25,
  status: "success",
  fileStatus: "success"
};

const onClickCallback = jest.fn();

describe("File withStyles - Invalid File", () => {
  wrapper = mount(
    <HvProvider>
      <FileWithStyles
        data={dataFail}
        onFilesAdded={() => {}}
        onFileRemoved={() => {}}
        progressConjunctionLabel="progressConjunctionLabel"
        removeFileButtonLabel="progressConjunctionLabel"
      />
    </HvProvider>
  );

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the File", () => {
    const file = wrapper.find("File");
    expect(file.length).toBe(1);
  });
});

describe("File withStyles - Valid File", () => {
  wrapper = mount(
    <HvProvider>
      <FileWithStyles
        data={dataSuccess}
        unit="mb"
        onFilesAdded={() => {}}
        onFileRemoved={onClickCallback}
        progressConjunctionLabel="progressConjunctionLabel"
        removeFileButtonLabel="progressConjunctionLabel"
      />
    </HvProvider>
  );

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the File", () => {
    const file = wrapper.find("File");
    expect(file.length).toBe(1);
  });

  it("should call the delete file callback", () => {
    const fileButton = wrapper.find("button");

    fileButton.simulate("click");
    expect(onClickCallback).toHaveBeenCalled();
  });
});
