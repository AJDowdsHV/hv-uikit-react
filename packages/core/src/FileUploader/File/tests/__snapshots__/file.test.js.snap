// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[v3] File withStyles - Invalid File should render correctly 1`] = `
<WithStyles(File)
  data={
    Object {
      "errorMessage": "File type not allowed for upload",
      "fileStatus": "success",
      "name": "somefile.png",
      "progress": 25,
      "size": 875,
      "status": "success",
    }
  }
  onFileRemoved={[MockFunction]}
  onFilesAdded={[Function]}
  removeFileButtonLabel="removeFileButtonLabel"
  unit="mb"
>
  <File
    classes={
      Object {
        "fail": "HvFileUploaderFile-fail",
        "icon": "HvFileUploaderFile-icon",
        "nameText": "HvFileUploaderFile-nameText",
        "progressTextContainer": "HvFileUploaderFile-progressTextContainer",
        "progressbar": "HvFileUploaderFile-progressbar",
        "progressbarBack": "HvFileUploaderFile-progressbarBack",
        "removeButton": "HvFileUploaderFile-removeButton",
      }
    }
    data={
      Object {
        "errorMessage": "File type not allowed for upload",
        "fileStatus": "success",
        "name": "somefile.png",
        "progress": 25,
        "size": 875,
        "status": "success",
      }
    }
    onFileRemoved={[MockFunction]}
    onFilesAdded={[Function]}
    removeFileButtonLabel="removeFileButtonLabel"
    unit="mb"
  >
    <li>
      <Success
        className="HvFileUploaderFile-icon"
        color="sema1"
      >
        <HvIconBase
          className="HvFileUploaderFile-icon"
          iconSize="S"
          name="Success"
        >
          <div
            className="HvFileUploaderFile-icon HvIconBase-root HvIconBase-s"
            name="Success"
          >
            <svg
              focusable={false}
              height={16}
              viewBox="0 0 16 16"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                className="color0"
                d="M6.387 12L4 9.613l.707-.707 1.613 1.613 4.65-5.58.768.641zM8 15a6.953 6.953 0 01-4.943-2.057A7.023 7.023 0 118 15m8-7a8 8 0 10-8 8 8 8 0 008-8z"
                fill="#59941B"
              />
            </svg>
          </div>
        </HvIconBase>
      </Success>
      <WithStyles(ForwardRef)
        className="HvFileUploaderFile-nameText"
        noWrap={true}
        variant="highlightText"
      >
        <ForwardRef
          className="HvFileUploaderFile-nameText"
          classes={
            Object {
              "3xlTitle": "HvTypography-3xlTitle",
              "4xlTitle": "HvTypography-4xlTitle",
              "5xlTitle": "HvTypography-5xlTitle",
              "disabledText": "HvTypography-disabledText",
              "highlightText": "HvTypography-highlightText",
              "lTitle": "HvTypography-lTitle",
              "link": "HvTypography-link",
              "mTitle": "HvTypography-mTitle",
              "noWrap": "HvTypography-noWrap",
              "normalText": "HvTypography-normalText",
              "placeholderText": "HvTypography-placeholderText",
              "root": "HvTypography-root",
              "sTitle": "HvTypography-sTitle",
              "sectionTitle": "HvTypography-sectionTitle",
              "selectedNavText": "HvTypography-selectedNavText",
              "vizText": "HvTypography-vizText",
              "vizTextDisabled": "HvTypography-vizTextDisabled",
              "xlTitle": "HvTypography-xlTitle",
              "xsInlineLink": "HvTypography-xsInlineLink",
              "xsTitle": "HvTypography-xsTitle",
              "xxlTitle": "HvTypography-xxlTitle",
              "xxsTitle": "HvTypography-xxsTitle",
            }
          }
          noWrap={true}
          variant="highlightText"
        >
          <p
            className="HvFileUploaderFile-nameText HvTypography-root HvTypography-highlightText HvTypography-noWrap"
          >
            somefile.png
          </p>
        </ForwardRef>
      </WithStyles(ForwardRef)>
      <span
        className="HvFileUploaderFile-progressTextContainer"
      >
         | 
        <WithStyles(ForwardRef)
          variant="highlightText"
        >
          <ForwardRef
            classes={
              Object {
                "3xlTitle": "HvTypography-3xlTitle",
                "4xlTitle": "HvTypography-4xlTitle",
                "5xlTitle": "HvTypography-5xlTitle",
                "disabledText": "HvTypography-disabledText",
                "highlightText": "HvTypography-highlightText",
                "lTitle": "HvTypography-lTitle",
                "link": "HvTypography-link",
                "mTitle": "HvTypography-mTitle",
                "noWrap": "HvTypography-noWrap",
                "normalText": "HvTypography-normalText",
                "placeholderText": "HvTypography-placeholderText",
                "root": "HvTypography-root",
                "sTitle": "HvTypography-sTitle",
                "sectionTitle": "HvTypography-sectionTitle",
                "selectedNavText": "HvTypography-selectedNavText",
                "vizText": "HvTypography-vizText",
                "vizTextDisabled": "HvTypography-vizTextDisabled",
                "xlTitle": "HvTypography-xlTitle",
                "xsInlineLink": "HvTypography-xsInlineLink",
                "xsTitle": "HvTypography-xsTitle",
                "xxlTitle": "HvTypography-xxlTitle",
                "xxsTitle": "HvTypography-xxsTitle",
              }
            }
            variant="highlightText"
          >
            <p
              className="HvTypography-root HvTypography-highlightText"
            >
              25.00B
            </p>
          </ForwardRef>
        </WithStyles(ForwardRef)>
        <WithStyles(ForwardRef)>
          <ForwardRef
            classes={
              Object {
                "3xlTitle": "HvTypography-3xlTitle",
                "4xlTitle": "HvTypography-4xlTitle",
                "5xlTitle": "HvTypography-5xlTitle",
                "disabledText": "HvTypography-disabledText",
                "highlightText": "HvTypography-highlightText",
                "lTitle": "HvTypography-lTitle",
                "link": "HvTypography-link",
                "mTitle": "HvTypography-mTitle",
                "noWrap": "HvTypography-noWrap",
                "normalText": "HvTypography-normalText",
                "placeholderText": "HvTypography-placeholderText",
                "root": "HvTypography-root",
                "sTitle": "HvTypography-sTitle",
                "sectionTitle": "HvTypography-sectionTitle",
                "selectedNavText": "HvTypography-selectedNavText",
                "vizText": "HvTypography-vizText",
                "vizTextDisabled": "HvTypography-vizTextDisabled",
                "xlTitle": "HvTypography-xlTitle",
                "xsInlineLink": "HvTypography-xsInlineLink",
                "xsTitle": "HvTypography-xsTitle",
                "xxlTitle": "HvTypography-xxlTitle",
                "xxsTitle": "HvTypography-xxsTitle",
              }
            }
          >
            <p
              className="HvTypography-root HvTypography-normalText"
            >
               / 875.00B
            </p>
          </ForwardRef>
        </WithStyles(ForwardRef)>
      </span>
      <WithStyles(ForwardRef(IconButton))
        aria-label="removeFileButtonLabel"
        category="ghost"
        className="HvFileUploaderFile-removeButton"
        onClick={[Function]}
      >
        <ForwardRef(IconButton)
          aria-label="removeFileButtonLabel"
          category="ghost"
          className="HvFileUploaderFile-removeButton"
          classes={
            Object {
              "colorInherit": "MuiIconButton-colorInherit",
              "colorPrimary": "MuiIconButton-colorPrimary",
              "colorSecondary": "MuiIconButton-colorSecondary",
              "disabled": "Mui-disabled",
              "edgeEnd": "MuiIconButton-edgeEnd",
              "edgeStart": "MuiIconButton-edgeStart",
              "label": "MuiIconButton-label",
              "root": "MuiIconButton-root",
              "sizeSmall": "MuiIconButton-sizeSmall",
            }
          }
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(ButtonBase))
            aria-label="removeFileButtonLabel"
            category="ghost"
            centerRipple={true}
            className="MuiIconButton-root HvFileUploaderFile-removeButton"
            disabled={false}
            focusRipple={true}
            onClick={[Function]}
          >
            <ForwardRef(ButtonBase)
              aria-label="removeFileButtonLabel"
              category="ghost"
              centerRipple={true}
              className="MuiIconButton-root HvFileUploaderFile-removeButton"
              classes={
                Object {
                  "disabled": "Mui-disabled",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiButtonBase-root",
                }
              }
              disableRipple={true}
              disabled={false}
              focusRipple={true}
              onClick={[Function]}
            >
              <button
                aria-label="removeFileButtonLabel"
                category="ghost"
                className="MuiButtonBase-root MuiIconButton-root HvFileUploaderFile-removeButton"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <Close
                    iconSize="XS"
                  >
                    <HvIconBase
                      iconSize="XS"
                      name="Close"
                    >
                      <div
                        className="HvIconBase-root HvIconBase-xs"
                        name="Close"
                      >
                        <svg
                          focusable={false}
                          height={12}
                          viewBox="0 0 16 16"
                          width={12}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            className="color0"
                            d="M8.7 8l5.3 5.3-.7.7L8 8.7 2.7 14l-.7-.7L7.3 8 2 2.7l.7-.7L8 7.3 13.3 2l.7.7z"
                            fill="#414141"
                          />
                        </svg>
                      </div>
                    </HvIconBase>
                  </Close>
                </span>
              </button>
            </ForwardRef(ButtonBase)>
          </WithStyles(ForwardRef(ButtonBase))>
        </ForwardRef(IconButton)>
      </WithStyles(ForwardRef(IconButton))>
    </li>
  </File>
</WithStyles(File)>
`;

exports[`[v3] File withStyles - Valid File should render correctly 1`] = `
<WithStyles(File)
  data={
    Object {
      "errorMessage": "File type not allowed for upload",
      "fileStatus": "success",
      "name": "somefile.png",
      "progress": 25,
      "size": 875,
      "status": "success",
    }
  }
  onFileRemoved={[MockFunction]}
  onFilesAdded={[Function]}
  removeFileButtonLabel="removeFileButtonLabel"
  unit="mb"
>
  <File
    classes={
      Object {
        "fail": "HvFileUploaderFile-fail",
        "icon": "HvFileUploaderFile-icon",
        "nameText": "HvFileUploaderFile-nameText",
        "progressTextContainer": "HvFileUploaderFile-progressTextContainer",
        "progressbar": "HvFileUploaderFile-progressbar",
        "progressbarBack": "HvFileUploaderFile-progressbarBack",
        "removeButton": "HvFileUploaderFile-removeButton",
      }
    }
    data={
      Object {
        "errorMessage": "File type not allowed for upload",
        "fileStatus": "success",
        "name": "somefile.png",
        "progress": 25,
        "size": 875,
        "status": "success",
      }
    }
    onFileRemoved={[MockFunction]}
    onFilesAdded={[Function]}
    removeFileButtonLabel="removeFileButtonLabel"
    unit="mb"
  >
    <li>
      <Success
        className="HvFileUploaderFile-icon"
        color="sema1"
      >
        <HvIconBase
          className="HvFileUploaderFile-icon"
          iconSize="S"
          name="Success"
        >
          <div
            className="HvFileUploaderFile-icon HvIconBase-root HvIconBase-s"
            name="Success"
          >
            <svg
              focusable={false}
              height={16}
              viewBox="0 0 16 16"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                className="color0"
                d="M6.387 12L4 9.613l.707-.707 1.613 1.613 4.65-5.58.768.641zM8 15a6.953 6.953 0 01-4.943-2.057A7.023 7.023 0 118 15m8-7a8 8 0 10-8 8 8 8 0 008-8z"
                fill="#59941B"
              />
            </svg>
          </div>
        </HvIconBase>
      </Success>
      <WithStyles(ForwardRef)
        className="HvFileUploaderFile-nameText"
        noWrap={true}
        variant="highlightText"
      >
        <ForwardRef
          className="HvFileUploaderFile-nameText"
          classes={
            Object {
              "3xlTitle": "HvTypography-3xlTitle",
              "4xlTitle": "HvTypography-4xlTitle",
              "5xlTitle": "HvTypography-5xlTitle",
              "disabledText": "HvTypography-disabledText",
              "highlightText": "HvTypography-highlightText",
              "lTitle": "HvTypography-lTitle",
              "link": "HvTypography-link",
              "mTitle": "HvTypography-mTitle",
              "noWrap": "HvTypography-noWrap",
              "normalText": "HvTypography-normalText",
              "placeholderText": "HvTypography-placeholderText",
              "root": "HvTypography-root",
              "sTitle": "HvTypography-sTitle",
              "sectionTitle": "HvTypography-sectionTitle",
              "selectedNavText": "HvTypography-selectedNavText",
              "vizText": "HvTypography-vizText",
              "vizTextDisabled": "HvTypography-vizTextDisabled",
              "xlTitle": "HvTypography-xlTitle",
              "xsInlineLink": "HvTypography-xsInlineLink",
              "xsTitle": "HvTypography-xsTitle",
              "xxlTitle": "HvTypography-xxlTitle",
              "xxsTitle": "HvTypography-xxsTitle",
            }
          }
          noWrap={true}
          variant="highlightText"
        >
          <p
            className="HvFileUploaderFile-nameText HvTypography-root HvTypography-highlightText HvTypography-noWrap"
          >
            somefile.png
          </p>
        </ForwardRef>
      </WithStyles(ForwardRef)>
      <span
        className="HvFileUploaderFile-progressTextContainer"
      >
         | 
        <WithStyles(ForwardRef)
          variant="highlightText"
        >
          <ForwardRef
            classes={
              Object {
                "3xlTitle": "HvTypography-3xlTitle",
                "4xlTitle": "HvTypography-4xlTitle",
                "5xlTitle": "HvTypography-5xlTitle",
                "disabledText": "HvTypography-disabledText",
                "highlightText": "HvTypography-highlightText",
                "lTitle": "HvTypography-lTitle",
                "link": "HvTypography-link",
                "mTitle": "HvTypography-mTitle",
                "noWrap": "HvTypography-noWrap",
                "normalText": "HvTypography-normalText",
                "placeholderText": "HvTypography-placeholderText",
                "root": "HvTypography-root",
                "sTitle": "HvTypography-sTitle",
                "sectionTitle": "HvTypography-sectionTitle",
                "selectedNavText": "HvTypography-selectedNavText",
                "vizText": "HvTypography-vizText",
                "vizTextDisabled": "HvTypography-vizTextDisabled",
                "xlTitle": "HvTypography-xlTitle",
                "xsInlineLink": "HvTypography-xsInlineLink",
                "xsTitle": "HvTypography-xsTitle",
                "xxlTitle": "HvTypography-xxlTitle",
                "xxsTitle": "HvTypography-xxsTitle",
              }
            }
            variant="highlightText"
          >
            <p
              className="HvTypography-root HvTypography-highlightText"
            >
              25.00B
            </p>
          </ForwardRef>
        </WithStyles(ForwardRef)>
        <WithStyles(ForwardRef)>
          <ForwardRef
            classes={
              Object {
                "3xlTitle": "HvTypography-3xlTitle",
                "4xlTitle": "HvTypography-4xlTitle",
                "5xlTitle": "HvTypography-5xlTitle",
                "disabledText": "HvTypography-disabledText",
                "highlightText": "HvTypography-highlightText",
                "lTitle": "HvTypography-lTitle",
                "link": "HvTypography-link",
                "mTitle": "HvTypography-mTitle",
                "noWrap": "HvTypography-noWrap",
                "normalText": "HvTypography-normalText",
                "placeholderText": "HvTypography-placeholderText",
                "root": "HvTypography-root",
                "sTitle": "HvTypography-sTitle",
                "sectionTitle": "HvTypography-sectionTitle",
                "selectedNavText": "HvTypography-selectedNavText",
                "vizText": "HvTypography-vizText",
                "vizTextDisabled": "HvTypography-vizTextDisabled",
                "xlTitle": "HvTypography-xlTitle",
                "xsInlineLink": "HvTypography-xsInlineLink",
                "xsTitle": "HvTypography-xsTitle",
                "xxlTitle": "HvTypography-xxlTitle",
                "xxsTitle": "HvTypography-xxsTitle",
              }
            }
          >
            <p
              className="HvTypography-root HvTypography-normalText"
            >
               / 875.00B
            </p>
          </ForwardRef>
        </WithStyles(ForwardRef)>
      </span>
      <WithStyles(ForwardRef(IconButton))
        aria-label="removeFileButtonLabel"
        category="ghost"
        className="HvFileUploaderFile-removeButton"
        onClick={[Function]}
      >
        <ForwardRef(IconButton)
          aria-label="removeFileButtonLabel"
          category="ghost"
          className="HvFileUploaderFile-removeButton"
          classes={
            Object {
              "colorInherit": "MuiIconButton-colorInherit",
              "colorPrimary": "MuiIconButton-colorPrimary",
              "colorSecondary": "MuiIconButton-colorSecondary",
              "disabled": "Mui-disabled",
              "edgeEnd": "MuiIconButton-edgeEnd",
              "edgeStart": "MuiIconButton-edgeStart",
              "label": "MuiIconButton-label",
              "root": "MuiIconButton-root",
              "sizeSmall": "MuiIconButton-sizeSmall",
            }
          }
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(ButtonBase))
            aria-label="removeFileButtonLabel"
            category="ghost"
            centerRipple={true}
            className="MuiIconButton-root HvFileUploaderFile-removeButton"
            disabled={false}
            focusRipple={true}
            onClick={[Function]}
          >
            <ForwardRef(ButtonBase)
              aria-label="removeFileButtonLabel"
              category="ghost"
              centerRipple={true}
              className="MuiIconButton-root HvFileUploaderFile-removeButton"
              classes={
                Object {
                  "disabled": "Mui-disabled",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiButtonBase-root",
                }
              }
              disableRipple={true}
              disabled={false}
              focusRipple={true}
              onClick={[Function]}
            >
              <button
                aria-label="removeFileButtonLabel"
                category="ghost"
                className="MuiButtonBase-root MuiIconButton-root HvFileUploaderFile-removeButton"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <Close
                    iconSize="XS"
                  >
                    <HvIconBase
                      iconSize="XS"
                      name="Close"
                    >
                      <div
                        className="HvIconBase-root HvIconBase-xs"
                        name="Close"
                      >
                        <svg
                          focusable={false}
                          height={12}
                          viewBox="0 0 16 16"
                          width={12}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            className="color0"
                            d="M8.7 8l5.3 5.3-.7.7L8 8.7 2.7 14l-.7-.7L7.3 8 2 2.7l.7-.7L8 7.3 13.3 2l.7.7z"
                            fill="#414141"
                          />
                        </svg>
                      </div>
                    </HvIconBase>
                  </Close>
                </span>
              </button>
            </ForwardRef(ButtonBase)>
          </WithStyles(ForwardRef(ButtonBase))>
        </ForwardRef(IconButton)>
      </WithStyles(ForwardRef(IconButton))>
    </li>
  </File>
</WithStyles(File)>
`;
